ee/clickhouse/queries/funnels/base.py

```diff
@@ -362,6 +362,7 @@ def _get_inner_event_query(
             team=self._team,
             extra_fields=[*self._extra_event_fields, *extra_fields],
             extra_event_properties=self._extra_event_properties,
+            using_person_on_events=self._team.actor_on_events_querying_enabled,
```

Comment:
@neilkakkar Not super sure what you had in mind here with this pattern. Should this be passed through all the different funnelquery types? Or is this right?

The problem I think that will occur with this is an implicit dependency. For example, in `ClickhouseTrends`, the class doesn't accept `using_person_on_events`but if the team has it enabled, `ClickhouseTrendsBreakdown` will automatically apply the flag. This seems difficult to trace

Reply:
Very valid point!

The thing is:

1. This flag is temporary
2. Keeping it at the top level means we need to update all tests + ensure we don't miss any paths where we also explicitly have to set this, making test coverage harder.

Keeping these two in mind, I found implicit to be the best choice: where, if this flag is switched on, everywhere should start using person-on-events. We _shouldn't_ have to think about which place is using it, and which place isn't. It's the whole thing.

This final thing: That it's the entire funnel insights slice that together starts using person-on-events, made being explicit very cumbersome, and not worth it imo. Same for trends, it's everything (not just breakdowns) that start using person-on-events when enabled. The param is there for now because Trends, Funnels, other insights, all actually have common code (EnterpriseEventQuery), where we, for now, need the flag to determine what to do.

Once all of these are sorted, we can directly pass the team variable everywhere. 

As always, open to other ways of doing things :)

Reply:
I've added the removal to clean up tasks: https://github.com/PostHog/posthog/issues/9802